model/src/main/kotlin/config/ProviderPluginConfiguration.kt

```diff
@@ -64,9 +64,7 @@ data class ProviderPluginConfiguration(
     @JsonIgnore
     val secrets: Options = emptyMap()
 ) {
-    override fun toString(): String {
-        // Do not use the generated toString function for the data class to ensure that the output does not contain the
-        // secret options.
-        return "${this::class.simpleName}(type=$type, id=$id, enabled=$enabled, options=$options)"
-    }
+    // Do not use the generated toString function for the data class to ensure that the output does not contain the
+    // secret options.
```

Comment:
Nit: How about also rephrasing this slightly into a proper function documentation now?